errdefs/http_helpers.go

```diff
@@ -5,6 +5,8 @@ import (
 )
 
 // FromStatusCode creates an errdef error, based on the provided HTTP status-code
+//
+// Deprecated: Use [cerrdefs.ToNative] instead
 func FromStatusCode(err error, statusCode int) error {
```

Comment:
Can we make this wrap `httpErrorFromStatusCode`? That way the test would also cover that the behavior is un-modified.

Reply:
That would require moving it out of `client` to a shared package though. Possibly under `internal`, but that would make the `errdefs` depending on the `internal`.. (although that might not be such a big deal if it's going away?)

Reply:
oh! my mistake; I glanced over after I commented, and thought they were in the same package 😞